service/src/main/java/uk/gov/hmcts/reform/fpl/service/CommonCaseDataExtractionService.java

```diff
@@ -53,6 +64,29 @@ public String extractPrehearingAttendance(HearingBooking booking) {
         return booking.hasDatesOnSameDay() ? formatTime(time) : formatDateTimeWithYear(time);
     }
 
+    public Map<String, Object> getHearingBookingData(final HearingBooking hearingBooking) {
```

Comment:
May I ask you not to operate on maps? They are harder to refactor and encourage to bad code practices.

Reply:
The one in `CaseDataExtractionService` returns a map which also contains information about the judge and legal advisor where as this one doesn't

Reply:
In our situation it would have overridden existing entires in the map with incorrect data